tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/Common/MatterProvision.cs

```diff
@@ -821,15 +831,34 @@ internal static bool CopyRoleAssignment(string conflictIdentified, string matter
         /// <returns>Matter details from matter library property bag</returns>
         internal MatterDetails ExtractMatterDetails(Dictionary<string, object> stampedPropertyValues)
         {
-            MatterDetails matterDetails = new MatterDetails()
+            MatterDetails matterDetails = new MatterDetails();
+
+            matterDetails.ResponsibleAttorney = GetStampPropertyValue(stampedPropertyValues, matterSettings.StampedPropertyResponsibleAttorney);
+            matterDetails.TeamMembers = GetStampPropertyValue(stampedPropertyValues, matterSettings.StampedPropertyTeamMembers);
+            matterDetails.UploadBlockedUsers = 
+                GetStampPropertyValue(stampedPropertyValues, matterSettings.StampedPropertyBlockedUploadUsers).Split(new string[] { ServiceConstants.SEMICOLON }, 
+                StringSplitOptions.RemoveEmptyEntries).ToList();
+
+
+            /*
+             * All the managed columns need to be read from the appsettings.json file. In old implementation
+             * all the managed columns are hardcoded and that hardcoding has been removed, by reading the
+             * column names from appsettings.json file
+             */
+            //Get the number of levels from Taxonomy Settings
+            int levels = int.Parse(configuration.GetSection("Taxonomy")["Levels"].ToString());
+            IDictionary<string, ManagedColumn> managedColumns = new Dictionary<string, ManagedColumn>();
+            for (int i = 1; i <= levels; i++)
             {
-                PracticeGroup = GetStampPropertyValue(stampedPropertyValues, matterSettings.StampedPropertyPracticeGroup),

```

Comment:
PracticeGroup, AreaOfLaw, SubareaOfLaw will be dynamic and cannot be hard coded. Now the code has been made generic to handle dynamic taxonomy hierarchy